pkg/machines/components/vmOverviewTabLibvirt.jsx

```diff
@@ -63,15 +68,36 @@ class VmOverviewTabLibvirt extends React.Component {
             runningVmUpdated: false,
             showVcpuModal: false,
             showBootOrderModal: false,
-            showMemoryModal: false
+            showMemoryModal: false,
+            showFirmwareModal: false,
         };
         this.openVcpu = this.openVcpu.bind(this);
         this.openMemory = this.openMemory.bind(this);
         this.openBootOrder = this.openBootOrder.bind(this);
+        this.openFirmware = this.openFirmware.bind(this);
         this.close = this.close.bind(this);
         this.onAutostartChanged = this.onAutostartChanged.bind(this);
     }
 
+    componentDidMount() {
+        if (this.props.config.provider.name == 'LibvirtDBus') {
+            getDomainCapabilities(this.props.vm.connectionName)
+                    .done(domCaps => {
+                        const parser = new DOMParser();
+                        const xmlDoc = parser.parseFromString(domCaps, "application/xml");
+                        if (!xmlDoc)
+                            return;
+
+                        const domainCapabilities = xmlDoc.getElementsByTagName("domainCapabilities")[0];
```

Comment:
Maybe I should check here before accessing the elements,  not clear if the always exist in the XML.

Reply:
According to libvirt's `docs/schemas/domaincaps.rng`, the `domainCapabilities` element is not optional, so it will always be there